libs/utils/executor.py

```diff
@@ -464,38 +473,38 @@ def _wload_init(self, tc, wl_idx):
         wload = self._wload_conf(wl_idx, wlspec)
 
         # Keep track of platform configuration
-        self.te.test_dir = '{}/{}:{}:{}'\
-            .format(self.te.res_dir, wload.wtype, tc_idx, wl_idx)
-        os.system('mkdir -p ' + self.te.test_dir)
-        self.te.platform_dump(self.te.test_dir)
+        test_dir = '{}/{}:{}:{}'.format(self.te.res_dir, wload.wtype,
+                                        tc_idx, wl_idx)
+        os.system('mkdir -p ' + test_dir)
+        self.te.platform_dump(test_dir)
 
         # Keep track of kernel configuration and version
         config = self.target.config
-        with gzip.open(os.path.join(self.te.test_dir, 'kernel.config'), 'wb') as fh:
+        with gzip.open(os.path.join(test_dir, 'kernel.config'), 'wb') as fh:
             fh.write(config.text)
         output = self.target.execute('{} uname -a'\
                 .format(self.target.busybox))
-        with open(os.path.join(self.te.test_dir, 'kernel.version'), 'w') as fh:
+        with open(os.path.join(test_dir, 'kernel.version'), 'w') as fh:
             fh.write(output)
 
-        return wload
+        return wload, test_dir
 
-    def _wload_run_init(self, run_idx):
-        self.te.out_dir = '{}/{}'\
-                .format(self.te.test_dir, run_idx)
-        logging.debug(r'%14s - out_dir [%s]', 'Executor', self.te.out_dir)
-        os.system('mkdir -p ' + self.te.out_dir)
+    def _wload_run_init(self, experiment):
+        experiment['out_dir'] = _get_experiment_out_dir(experiment)
 
-    def _wload_run(self, exp_idx, tc, wl_idx, wload, run_idx):
+    def _wload_run(self, exp_idx, experiment):
+        tc = experiment['conf']
+        wload = experiment['wload']
         tc_idx = tc['tag']
 
         self._print_title('Executor', 'Experiment {}/{}, [{}:{}] {}/{}'\
-                .format(exp_idx, self._exp_count,
-                        tc_idx, wl_idx,
-                        run_idx, self._iterations))
+                .format(exp_idx, self._exp_count, # todo len(self.experiments?)
```

Comment:
Removed the todo; it would be nice to get rid of the calculation of `self._exp_count` in `__init__` but for now I think it's more trouble than it's worth.

Re: s/experiment/exp/: why? We aren't short on horizontal space (the `format` call has to be broken up, but it would be anyway), so why not be totally clear on what the variable refers to?